torch/_torch_docs.py

```diff
@@ -3587,6 +3587,54 @@ def merge_dicts(*dicts):
     array([-1,  2,  3])
 """)
 
+add_docstr(torch.frombuffer,
+           r"""
+frombuffer(buffer, dtype=None, count=-1, offset=0, *, requires_grad=False) -> Tensor
+
+Create a 1-dimensional :class:`Tensor` from an object that implements
+the Python buffer protocol.
+
+Interpret :attr:`buffer` as a 1-dimensional buffer of type :attr:`dtype`
+with :attr:`count` elements after the first :attr:`offset` bytes.
+
+The returned tensor and buffer share the same memory. Modifications to
+the tensor will be reflected in the buffer and vice versa. The returned
+tensor is not resizable.
+
+.. note::
+    This function increments the reference count for the object that
+    owns the shared memory. Therefore, such memory will not be deallocated
+    before the returned tensor goes out of scope.
+
+.. warning::
+    The buffer protocols has no notion of devices. Thus, this function only
```

Comment:
How could a CUDA buffer occur? Is there any way to detect this case or protect ourselves if it happens? Segfaults are bad.

Reply:
As far as I know, there is [`pycuda`](https://documen.tician.de/pycuda/tutorial.html) that provides CUDA buffers, and `mpi4py` that can use CUDA buffers ([link @rgommers referenced](https://gist.github.com/lebedov/8514d3456a94a6c73e6d)).

I am not sure how we could detect CUDA buffers, though. That is because, it all boils down to being able to tell CUDA and CPU pointers apart. AFAIK, it's something hard to be done, since a given pointer can be valid on both of them...

Reply:
Thanks for the explanation. I guess we'll stick with this for now